app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/room/db/dao/VideoDao.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (C) 2017 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package android.support.v17.leanback.supportleanbackshowcase.app.room.db.dao;
+
+import android.arch.lifecycle.LiveData;
+import android.arch.persistence.room.Dao;
+import android.arch.persistence.room.Insert;
+import android.arch.persistence.room.OnConflictStrategy;
+import android.arch.persistence.room.Query;
+import android.arch.persistence.room.Update;
+import android.support.v17.leanback.supportleanbackshowcase.app.room.db.constant.DatabaseContract;
+import android.support.v17.leanback.supportleanbackshowcase.app.room.db.entity.VideoEntity;
+
+import java.util.List;
+
+@Dao
+public interface VideoDao {
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
```

Comment:
i see, this is where you are using the constants.
you can keep it if you like but since Room verifies these queries at compile time, there is no reason to create these constants. if there is a mistake, room will throw a compilation error.

Reply:
Yes, that's a huge benefits comes along with Room. The class name has been updated for clarification.